david/src/day_001/day_001.go

```diff
@@ -0,0 +1,130 @@
+// Creator: david
+// Date: 2021-09-30
+// Purpose: day 1 of advent of code 2021
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+func txtFileReader(txtFile string) []string {
+	content, error := ioutil.ReadFile(txtFile)
+	if error != nil {
+		log.Fatal(error)
+	}
+	contentArray := strings.Split(string(content[:]), "\n")
+	return contentArray
+}
+
+var NUMBERS = map[string]int{
+	"zero": 0, "one": 1, "two": 2, "three": 3, "four": 4, "five": 5,
+	"six": 6, "seven": 7, "eight": 8, "nine": 9, "ten": 10,
+	"eleven": 11, "twelve": 12, "thirteen": 13, "fourteen": 14, "fifteen": 15,
+	"sixteen": 16, "seventeen": 17, "eighteen": 18, "nineteen": 19,
+	"twenty": 20, "thirty": 30, "forty": 40, "fifty": 50,
+	"sixty": 60, "seventy": 70, "eighty": 80, "ninety": 90,
+}
+
+// read txt file and return array of strings
+
+// reverse a string
+func reverse(s string) string {
+	runes := []rune(s)
+	for i, j := 0, len(runes)-1; i < j; i, j = i+1, j-1 {
+		runes[i], runes[j] = runes[j], runes[i]
```

Comment:
i'm actually kind of curious if this works correctly with things like diacritics and emoji. i don't know enough about unicode though